tests/integration.rs

```diff
@@ -29,12 +29,17 @@ mod integration_tests {
 
     const SPIN_BINARY: &str = "./target/debug/spin";
     const BINDLE_SERVER_BINARY: &str = "bindle-server";
+    const NOMAD_BINARY: &str = "nomad";
```

Comment:
With these two added dependencies for running `make test` locally, do you have thoughts on how best to alert users on their necessity?  Perhaps some combination of docs, obvious error/failure messages when a dep is missing and/or checks in the Makefile (e.g. as a first task when running `make test`, check for needed binaries and return non-zero with a helpful error message if any are missing)?

Reply:
I would like to use `type nomad` in `make test`. However, should we also consider Windows? If yes, it will be `type nomad.exe` on Windows. Not sure what is a proper way to do that on different platforms.

Reply:
I think we can use something like the following in the Makefile:

```Makefile
NOMAD := nomad
ifeq ($(OS),Windows_NT)
	NOMAD = nomad.exe
endif

nomad-version:
	$(NOMAD) --version
```

(`nomad-version` just example usage.)

Reply:
I added this part in `make test`, but not sure what is a proper way to test `Hippo.Web`. I tried `-h`, `--help`, `-v` and `--version`, but all show error.

Reply:
Hmmm, ok.  I don't have a Windows host right now to test, but would something like this work?

```Makefile
WHICH := which
EXT   :=
ifeq ($(OS),Windows_NT)
	WHICH = where
	EXT   = .exe
endif

.PHONY: ensure-bindle-server
ensure-bindle-server:
	@$(WHICH) bindle-server$(EXT) &>/dev/null || \
		(echo "bindle-server$(EXT) must be installed to run the integration tests" && exit 1)

.PHONY: ensure-hippo
ensure-hippo:
	@$(WHICH) Hippo.Web &>/dev/null || \
		(echo "Hippo.Web must be installed to run the integration tests" && exit 1)

.PHONY: ensure-nomad
ensure-nomad:
	@$(WHICH) nomad$(EXT) &>/dev/null || \
		(echo "nomad$(EXT) must be installed to run the integration tests" && exit 1)
```

I'm also open to ideas on where we should put these checks for the integration test dependencies.  Would it be better to place the checks in `integration.rs` itself?

Also, do we like continuing to run all tests (unit and integration) during `make test`?  Or do we want separate targets, e.g. `make test-unit`, `make test-integration`, etc.?  cc @michelleN

Reply:
I remove `$(OS)` in Makefile, so we don't need to test it in two OS. I add the dependency test to `integration.rs` and separate unit and integration test in Makefile.

Reply:
Nice, I like that the checks are in `integration.rs` now.  I did a quick search -- do we want to try using something like https://crates.io/crates/which which appears to also support Windows?

Also, can we add a helpful context message for when one of the dependencies isn't found?

Reply:
I change to use `which` and give it `with_context`. Thank you.

Reply:
Thanks!